controller/Controller.py

```diff
@@ -0,0 +1,207 @@
+    """DDS Controller class."""
+
+    def __init__(
+        self,
+        dds_domain=0,
+        debug=False,
+        logger=None
+    ):
+        """Construct a DDS Controller."""
+        super().__init__()
+        # Initialize logger
+        self.logger = Logger(debug, logger)
+        self.logger.debug('Initializing DDS Recorder controller...')
+
+        # Check DDS Domain
+        if (not self.is_valid_dds_domain(dds_domain)):
+            raise ValueError(
+                'DDS Domain must be a number between 0 and 255.')
+
+        self.init_dds(dds_domain)
+
+    def __del__(self):
+        """Remove all dds entities in an orderly manner."""
+        self.delete()
+
+    on_ddsrecorder_discovered = pyqtSignal(str)
+    on_ddsrecorder_status = pyqtSignal(str, str, str)
+
+    def is_valid_dds_domain(self, dds_domain):
+        return ((dds_domain >= 0) and (dds_domain <= 255))
+
+    def init_dds(self, dds_domain):
+        """Initialize DDS entities."""
+        factory = fastdds.DomainParticipantFactory.get_instance()
+        participant_qos = fastdds.DomainParticipantQos()
+        factory.get_default_participant_qos(participant_qos)
+        participant_qos.name('DDSRecorderController')
+        self.participant = factory.create_participant(
+            dds_domain, participant_qos)
+
+        # Initialize command topic
+        self.command_topic_data_type = ControllerCommandPubSubType()
+        self.command_topic_data_type.setName('ControllerCommand')
+        type_support = fastdds.TypeSupport(self.command_topic_data_type)
+        self.participant.register_type(type_support)
+
+        topic_qos = fastdds.TopicQos()
+        self.participant.get_default_topic_qos(topic_qos)
+        self.command_topic = self.participant.create_topic(
+            'ddsrecorder/controller',
+            self.command_topic_data_type.getName(),
+            topic_qos)
+
+        # Initialize status topic
+        self.status_topic_data_type = StatusPubSubType()
+        self.status_topic_data_type.setName('Status')
+        type_support = fastdds.TypeSupport(self.status_topic_data_type)
+        self.participant.register_type(type_support)
+
+        topic_qos = fastdds.TopicQos()
+        self.participant.get_default_topic_qos(topic_qos)
+        self.status_topic = self.participant.create_topic(
+            'ddsrecorder/status',
+            self.status_topic_data_type.getName(),
+            topic_qos)
+
+        publisher_qos = fastdds.PublisherQos()
+        self.participant.get_default_publisher_qos(publisher_qos)
+        self.publisher = self.participant.create_publisher(publisher_qos)
+
+        self.command_writer_listener = CommandWriterListener(self)
+        writer_qos = fastdds.DataWriterQos()
+        self.publisher.get_default_datawriter_qos(writer_qos)
+        writer_qos.reliability().kind = fastdds.RELIABLE_RELIABILITY_QOS
+        writer_qos.durability().kind = fastdds.TRANSIENT_LOCAL_DURABILITY_QOS
```

Comment:
Finally it is transient local? For a Late Joiner recorder, will perform every command in order (it may start and stop many times in a row in few milliseconds).

Reply:
@rsanchez15

Reply:
It will be reliable but not transient local